benchmarks/benchmarks/stats.py

```diff
@@ -174,3 +178,51 @@ def time_binned_statistic_dd_reuse_bin(self, statistic):
         stats.binned_statistic_dd(
             [self.inp[0], self.inp[1]], self.inp[2], statistic=statistic,
             binned_statistic_result=self.ret)
+
+
+class ContinuousFitAnalyticalMLEOverride(Benchmark):
+    # list of distributions to time
+    dists = ["pareto", "laplace", "beta", "ncf"]
+    # add custom values for rvs and fit, if desired, for any distribution:
+    # key should match name in dists and value should be list of loc, scale,
+    # and shapes
+    custom_input = {"beta": [1, 1, 1, 1]}
+    fnames = ['floc', 'fscale', 'f0', 'f1', 'f2']
+    fixed = {}
+
+    param_names = ["distribution", "loc_fixed", "scale_fixed",
+                   "shape1_fixed", "shape2_fixed", "shape3_fixed"]
+    params = [dists, * [[True, False]] * 5]
+
+    def setup(self, dist_name, loc_fixed, scale_fixed, shape1_fixed,
+              shape2_fixed, shape3_fixed):
+        exec("self.distn= stats." + dist_name)
+
+        nparam = (len(self.distn.shapes.replace(',', ' ').split())
```

Comment:
```suggestion
        nparam = (len(self.distn.shapes.split(", "))
```
Would this work?

Reply:
But probably better to form `default_shapes` (see below) up here, then just `len(default_shapes)`.

Reply:
For our purposes, yes. 

```
In [11]: shapes = 'a, b'                                                                          

In [12]: shapes.replace(',', ' ').split()                                                         
Out[12]: ['a', 'b']

In [13]: shapes.split()                                                                           
Out[13]: ['a,', 'b']

In [14]: len(shapes.split())                                                                      
Out[14]: 2

In [15]: len(shapes.replace(',', ' ').split())                                                    
Out[15]: 2
```